src/main/java/org/spongepowered/api/event/inventory/InventoryOpenEvent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.inventory;
+
+import org.spongepowered.api.util.event.Cancellable;
+
+/**
+ * Fired when a viewer opens an inventory.
+ */
+public interface InventoryOpenEvent extends ViewerEvent, Cancellable {
```

Comment:
@spaceemotion I don't know about the minecraft internals, but I'm *pretty* sure
bukkit plugins were able to do this

On Mon Jan 05 2015 at 8:23:44 AM spaceemotion <notifications@github.com>
wrote:

> In
> src/main/java/org/spongepowered/api/event/inventory/InventoryOpenEvent.java
> <https://github.com/SpongePowered/SpongeAPI/pull/242#discussion-diff-22462496>
> :
>
> > + * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
> > + * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
> > + * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> > + * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> > + * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> > + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> > + * THE SOFTWARE.
> > + */
> > +package org.spongepowered.api.event.inventory;
> > +
> > +import org.spongepowered.api.util.event.Cancellable;
> > +
> > +/**
> > + * Fired when a viewer opens an inventory.
> > + */
> > +public interface InventoryOpenEvent extends ViewerEvent, Cancellable {
>
> Would this be cancellable even if its the normal PlayerInventory? My guess
> is "no", but would be interesting to know.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpongePowered/SpongeAPI/pull/242/files#r22462496>.
>